interfaces.go

```diff
@@ -3,8 +3,8 @@ package merkleTree
 // StorageEngine specifies how to store and lookup merkle tree nodes
 // and roots.  You can use a DB like Dynamo or SQL to do this.
 type StorageEngine interface {
```

Comment:
That will be revealed based on unpacking the data.  It currently works in
the tester storage which is now a byte oriented interface so should more
closely model the disk.

On Wednesday, February 3, 2016, Andy Alness <notifications@github.com>
wrote:

> In interfaces.go
> <https://github.com/keybase/go-merkle-tree/pull/1#discussion_r51758099>:
>
> > @@ -3,8 +3,8 @@ package merkleTree
> >  // StorageEngine specifies how to store and lookup merkle tree nodes
> >  // and roots.  You can use a DB like Dynamo or SQL to do this.
> >  type StorageEngine interface {
>
> I'm trying to visualize how this is going to work for someone looking up
> the nodes. If Node isn't passed to StoreNode how will a user know if it's
> an iNode or a leaf node when looking it up?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/keybase/go-merkle-tree/pull/1/files#r51758099>.
>